### ISSUE TITLE
[master] Panic when trying to backup data

### ISSUE THREAD

**Issue by papylhomme** [influxdata/influxdb#3700]
Hello,

Using current master (deabf8d2a8e60e701c32302b569a1006ba378ee5) I get a panic when using the command ```influxd backup mybackup```

Full output is here https://gist.github.com/papylhomme/59b2b830a6b7bb6af195

**Comment by beckettsean** [comment]
Please open a [GitHub Issue](https://github.com/influxdb/influxdb/issues/new)
for this so the development team can investigate.

On Mon, Aug 17, 2015 at 11:59 AM, Michaël Lhomme <notifications@github.com>
wrote:

> Hello,
>
> Using current master (deabf8d
> <https://github.com/influxdb/influxdb/commit/deabf8d2a8e60e701c32302b569a1006ba378ee5>)
> I get a panic when using the command influxd backup mybackup
>
> Full output is here
> https://gist.github.com/papylhomme/59b2b830a6b7bb6af195
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/issues/3700>.
>



-- 
Sean Beckett
Director of Support and Professional Services
InfluxDB

**Comment by papylhomme** [comment]
Yep, I had just started the server on a tty for a clean output ;) It happens each time I'm trying it. 

```
8888888           .d888 888                   8888888b.  888888b.
888            d88P"  888                   888  "Y88b 888  "88b
888            888    888                   888    888 888  .88P
888   88888b.  888888 888 888  888 888  888 888    888 8888888K.
888   888 "88b 888    888 888  888  Y8bd8P' 888    888 888  "Y88b 
888   888  888 888    888 888  888   X88K   888    888 888    888 
888   888  888 888    888 Y88b 888 .d8""8b. 888  .d88P 888   d88P 
8888888 888  888 888    888  "Y88888 888  888 8888888P"  8888888P"
using configuration at: /etc/influxdb.conf
[continuous_querier] 2015/08/16 12:52:54 starting continuous query service
[metastore] 2015/08/16 12:52:54 read local node id: 1
[metastore] 2015/08/16 12:52:54 skipping join: already member of cluster: nodeId=1 raftEnabled=true raftNodes=[danska.home:8088]
[b1] 2015/08/16 12:52:54 flush 1221 points in 0.092s
[b1] 2015/08/16 12:52:54 flush 1268 points in 0.083s
[b1] 2015/08/16 12:52:55 flush 1077 points in 0.072s
[b1] 2015/08/16 12:52:55 flush 1078 points in 0.073s
[b1] 2015/08/16 12:52:55 flush 938 points in 0.061s
[b1] 2015/08/16 12:52:55 flush 1033 points in 0.069s
[b1] 2015/08/16 12:52:55 flush 986 points in 0.063s
[b1] 2015/08/16 12:52:55 flush 1033 points in 0.069s
[admin] 2015/08/16 12:52:55 listening on HTTP: 127.0.0.1:8083
[httpd] 2015/08/16 12:52:55 authentication enabled: false
[httpd] 2015/08/16 12:52:55 listening on HTTP: 127.0.0.1:8086
[metastore] 2015/08/16 12:52:55 updated node id=1 hostname=danska.home:8088
[collectd] 2015/08/16 12:52:55 collectd UDP started
2015/08/16 12:52:55 InfluxDB starting, version 0.9, commit unknown
2015/08/16 12:52:55 GOMAXPROCS set to 4
[run] 2015/08/16 12:52:55 listening for signals
2015/08/16 12:52:56 Sending anonymous usage statistics to m.influxdb.com
```

**Comment by tslater** [comment]
I have the same issue. I was on v0.9.3.x but then updated to InfluxDB v0.9.4.1. 
I'm on ubuntu.
Here is the log output:

````
[http] 2015/09/24 21:33:00 52.88.133.242 - read-write-user [24/Sep/2015:21:33:00 +0000] POST /write?database=jolt-sensor-beta&db=jolt-sensor-beta&p=%5BREDACTED%5D&precision=ms&u=read-write-user HTTP/1.1 204 0 - - d47a8fa9-6303-11e5-8002-000000000000 3.625029ms
[wal] 2015/09/24 21:33:05 Flush due to idle. Flushing 4 series with 4 points and 68 bytes from partition 1
[wal] 2015/09/24 21:33:05 write to index of partition 1 took 2.548271ms
[wal] 2015/09/24 21:33:05 Flush due to idle. Flushing 10 series with 10 points and 602 bytes from partition 1
[wal] 2015/09/24 21:33:05 write to index of partition 1 took 2.315175ms
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x79a708]

goroutine 79 [running]:
github.com/boltdb/bolt.(*DB).beginTx(0x0, 0xc820542d80, 0x0, 0x0)
	/tmp/build/src/github.com/boltdb/bolt/db.go:422 +0x2f8
github.com/boltdb/bolt.(*DB).Begin(0x0, 0x0, 0x7fa6ee10eae8, 0x0, 0x0)
	/tmp/build/src/github.com/boltdb/bolt/db.go:415 +0x60
github.com/influxdb/influxdb/tsdb.appendShardSnapshotFile(0xc820545d50, 0xc82013a6c0, 0xc820190ad7, 0x1a, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/tsdb/snapshot_writer.go:86 +0x1bd
github.com/influxdb/influxdb/tsdb.appendShardSnapshotFiles(0xc820545d50, 0xc820086620, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/tsdb/snapshot_writer.go:70 +0x669
github.com/influxdb/influxdb/tsdb.NewSnapshotWriter.func1(0xc820514800, 0x1f3, 0x200, 0xc820545d50, 0xc820086620, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/tsdb/snapshot_writer.go:31 +0x30d
github.com/influxdb/influxdb/tsdb.NewSnapshotWriter(0xc820514800, 0x1f3, 0x200, 0xc820086620, 0x0, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/tsdb/snapshot_writer.go:20 +0x122
github.com/influxdb/influxdb/services/snapshotter.(*Service).writeSnapshot(0xc82000b360, 0x7fa6ee116588, 0xc820029530, 0x0, 0x0, 0x0, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/services/snapshotter/service.go:131 +0x1c3
github.com/influxdb/influxdb/services/snapshotter.(*Service).handleConn(0xc82000b360, 0x7fa6ee116588, 0xc820029530, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/services/snapshotter/service.go:105 +0x1cd
github.com/influxdb/influxdb/services/snapshotter.(*Service).serve.func1(0xc82000b360, 0x7fa6ee116588, 0xc820029530)
	/tmp/build/src/github.com/influxdb/influxdb/services/snapshotter/service.go:89 +0xbf
created by github.com/influxdb/influxdb/services/snapshotter.(*Service).serve
	/tmp/build/src/github.com/influxdb/influxdb/services/snapshotter/service.go:92 +0x470

goroutine 1 [chan receive]:
main.(*Main).Run(0xc82026dec0, 0xc82000a0b0, 0x4, 0x4, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/cmd/influxd/main.go:91 +0xa16
main.main()
	/tmp/build/src/github.com/influxdb/influxdb/cmd/influxd/main.go:42 +0x29b

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/home/philip/.gvm/gos/go1.5/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [syscall]:
os/signal.loop()
	/home/philip/.gvm/gos/go1.5/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/home/philip/.gvm/gos/go1.5/src/os/signal/signal_unix.go:28 +0x37

goroutine 8 [IO wait]:
net.runtime_pollWait(0x7fa6ee110b10, 0x72, 0xc82000e1b0)
	/home/philip/.gvm/gos/go1.5/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82004f3a0, 0x72, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82004f3a0, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc82004f340, 0x0, 0x7fa6ee111160, 0xc82054a380)
	/home/philip/.gvm/gos/go1.5/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820028158, 0xc8202b4ec8, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/tcpsock_posix.go:254 +0x4d
net.(*TCPListener).Accept(0xc820028158, 0x0, 0x0, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/tcpsock_posix.go:264 +0x3d
github.com/influxdb/influxdb/tcp.(*Mux).Serve(0xc820124a40, 0x7fa6ee10fb90, 0xc820028158, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/tcp/mux.go:48 +0x4b
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Open.func1
	/tmp/build/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:345 +0xae9

goroutine 9 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc820028160, 0x0, 0x0, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/meta.(*raftLayer).Accept(0xc82012b830, 0x0, 0x0, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/meta/store.go:2059 +0x51
github.com/hashicorp/raft.(*NetworkTransport).listen(0xc82004f3b0)
	/tmp/build/src/github.com/hashicorp/raft/net_transport.go:346 +0x79
created by github.com/hashicorp/raft.NewNetworkTransport
	/tmp/build/src/github.com/hashicorp/raft/net_transport.go:138 +0x284

goroutine 10 [select]:
github.com/hashicorp/raft.(*Raft).leaderLoop(0xc82008c1c0)
	/tmp/build/src/github.com/hashicorp/raft/raft.go:849 +0xcce
github.com/hashicorp/raft.(*Raft).runLeader(0xc82008c1c0)
	/tmp/build/src/github.com/hashicorp/raft/raft.go:821 +0x936
github.com/hashicorp/raft.(*Raft).run(0xc82008c1c0)
	/tmp/build/src/github.com/hashicorp/raft/raft.go:602 +0xbb
github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/raft.run)-fm()
	/tmp/build/src/github.com/hashicorp/raft/raft.go:252 +0x20
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc82008c1c0, 0xc8201309a0)
	/tmp/build/src/github.com/hashicorp/raft/state.go:152 +0x44
created by github.com/hashicorp/raft.(*raftState).goFunc
	/tmp/build/src/github.com/hashicorp/raft/state.go:153 +0x4d

goroutine 11 [select]:
github.com/hashicorp/raft.(*Raft).runFSM(0xc82008c1c0)
	/tmp/build/src/github.com/hashicorp/raft/raft.go:509 +0x10eb
github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/raft.runFSM)-fm()
	/tmp/build/src/github.com/hashicorp/raft/raft.go:253 +0x20
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc82008c1c0, 0xc8201309b0)
	/tmp/build/src/github.com/hashicorp/raft/state.go:152 +0x44
created by github.com/hashicorp/raft.(*raftState).goFunc
	/tmp/build/src/github.com/hashicorp/raft/state.go:153 +0x4d

goroutine 12 [select]:
github.com/hashicorp/raft.(*Raft).runSnapshots(0xc82008c1c0)
	/tmp/build/src/github.com/hashicorp/raft/raft.go:1636 +0x493
github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/raft.runSnapshots)-fm()
	/tmp/build/src/github.com/hashicorp/raft/raft.go:254 +0x20
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc82008c1c0, 0xc8201309c0)
	/tmp/build/src/github.com/hashicorp/raft/state.go:152 +0x44
created by github.com/hashicorp/raft.(*raftState).goFunc
	/tmp/build/src/github.com/hashicorp/raft/state.go:153 +0x4d

goroutine 13 [select]:
github.com/influxdb/influxdb/meta.(*localRaft).logLeaderChanges(0xc82000b9f0)
	/tmp/build/src/github.com/influxdb/influxdb/meta/state.go:187 +0x4f5
created by github.com/influxdb/influxdb/meta.(*localRaft).open
	/tmp/build/src/github.com/influxdb/influxdb/meta/state.go:177 +0x1344

goroutine 14 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc820028168, 0x0, 0x0, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/meta.(*Store).serveExecListener(0xc82007c8c0)
	/tmp/build/src/github.com/influxdb/influxdb/meta/store.go:622 +0xa2
created by github.com/influxdb/influxdb/meta.(*Store).Open
	/tmp/build/src/github.com/influxdb/influxdb/meta/store.go:239 +0x364

goroutine 15 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc820028170, 0x0, 0x0, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/meta.(*Store).serveRPCListener(0xc82007c8c0)
	/tmp/build/src/github.com/influxdb/influxdb/meta/store.go:726 +0xa2
created by github.com/influxdb/influxdb/meta.(*Store).Open
	/tmp/build/src/github.com/influxdb/influxdb/meta/store.go:242 +0x3b6

goroutine 21 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc820086700, 0xc8201ea000)
	/tmp/build/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x567
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/tmp/build/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x627

goroutine 19 [select]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).monitorErrorChan(0xc8200f17a0, 0xc82001a900)
	/tmp/build/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:491 +0x145
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Open.func1
	/tmp/build/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:351 +0xc67

goroutine 20 [select]:
github.com/influxdb/influxdb/monitor.(*Monitor).storeStatistics(0xc8200991e0)
	/tmp/build/src/github.com/influxdb/influxdb/monitor/service.go:333 +0x187c
created by github.com/influxdb/influxdb/monitor.(*Monitor).Open
	/tmp/build/src/github.com/influxdb/influxdb/monitor/service.go:134 +0x30c

goroutine 22 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc8200867e0, 0xc82028ca80)
	/tmp/build/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x567
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/tmp/build/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x627

goroutine 23 [select]:
github.com/influxdb/influxdb/services/hh.(*Service).retryWrites(0xc8200745a0)
	/tmp/build/src/github.com/influxdb/influxdb/services/hh/service.go:103 +0x314
created by github.com/influxdb/influxdb/services/hh.(*Service).Open
	/tmp/build/src/github.com/influxdb/influxdb/services/hh/service.go:67 +0x210

goroutine 24 [select]:
github.com/influxdb/influxdb/services/hh.(*Service).expireWrites(0xc8200745a0)
	/tmp/build/src/github.com/influxdb/influxdb/services/hh/service.go:121 +0x2d1
created by github.com/influxdb/influxdb/services/hh.(*Service).Open
	/tmp/build/src/github.com/influxdb/influxdb/services/hh/service.go:68 +0x232

goroutine 25 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc820028178, 0x0, 0x0, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/cluster.(*Service).serve(0xc82004f2d0)
	/tmp/build/src/github.com/influxdb/influxdb/cluster/service.go:84 +0xdb
created by github.com/influxdb/influxdb/cluster.(*Service).Open
	/tmp/build/src/github.com/influxdb/influxdb/cluster/service.go:61 +0x135

goroutine 26 [select]:
github.com/influxdb/influxdb/services/precreator.(*Service).runPrecreation(0xc8201246c0)
	/tmp/build/src/github.com/influxdb/influxdb/services/precreator/service.go:75 +0x44f
created by github.com/influxdb/influxdb/services/precreator.(*Service).Open
	/tmp/build/src/github.com/influxdb/influxdb/services/precreator/service.go:53 +0x21f

goroutine 27 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc820028180, 0x0, 0x0, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/services/snapshotter.(*Service).serve(0xc82000b360)
	/tmp/build/src/github.com/influxdb/influxdb/services/snapshotter/service.go:75 +0xa9
created by github.com/influxdb/influxdb/services/snapshotter.(*Service).Open
	/tmp/build/src/github.com/influxdb/influxdb/services/snapshotter/service.go:48 +0x130

goroutine 28 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc820028188, 0x0, 0x0, 0x0, 0x0)
	/tmp/build/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/services/copier.(*Service).serve(0xc820124740)
	/tmp/build/src/github.com/influxdb/influxdb/services/copier/service.go:78 +0xa9
created by github.com/influxdb/influxdb/services/copier.(*Service).Open
	/tmp/build/src/github.com/influxdb/influxdb/services/copier/service.go:51 +0x130

goroutine 29 [IO wait]:
net.runtime_pollWait(0x7fa6ee110a50, 0x72, 0xc82000e1b0)
	/home/philip/.gvm/gos/go1.5/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820284370, 0x72, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820284370, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc820284310, 0x0, 0x7fa6ee111160, 0xc820294c40)
	/home/philip/.gvm/gos/go1.5/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820028df0, 0x455470, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/tcpsock_posix.go:254 +0x4d
net.(*TCPListener).Accept(0xc820028df0, 0x0, 0x0, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/tcpsock_posix.go:264 +0x3d
net/http.(*Server).Serve(0xc82028cfc0, 0x7fa6ee10fb90, 0xc820028df0, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/http/server.go:1887 +0xb3
net/http.Serve(0x7fa6ee10fb90, 0xc820028df0, 0x7fa6ee116198, 0xc82020bf80, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/http/server.go:1765 +0x85
github.com/influxdb/influxdb/services/admin.(*Service).serve(0xc82000b450)
	/tmp/build/src/github.com/influxdb/influxdb/services/admin/service.go:107 +0x12f
created by github.com/influxdb/influxdb/services/admin.(*Service).Open
	/tmp/build/src/github.com/influxdb/influxdb/services/admin/service.go:70 +0x491

goroutine 30 [select]:
github.com/influxdb/influxdb/services/continuous_querier.(*Service).backgroundLoop(0xc8200746c0)
	/tmp/build/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:192 +0x565
created by github.com/influxdb/influxdb/services/continuous_querier.(*Service).Open
	/tmp/build/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:126 +0x2a4

goroutine 31 [IO wait]:
net.runtime_pollWait(0x7fa6ee110990, 0x72, 0xc82000e1b0)
	/home/philip/.gvm/gos/go1.5/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8202843e0, 0x72, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8202843e0, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc820284380, 0x0, 0x7fa6ee111160, 0xc820500380)
	/home/philip/.gvm/gos/go1.5/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820028df8, 0xc820037dc0, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/tcpsock_posix.go:254 +0x4d
net.(*TCPListener).Accept(0xc820028df8, 0x0, 0x0, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/tcpsock_posix.go:264 +0x3d
net/http.(*Server).Serve(0xc82028d080, 0x7fa6ee10fb90, 0xc820028df8, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/http/server.go:1887 +0xb3
net/http.Serve(0x7fa6ee10fb90, 0xc820028df8, 0x7fa6ee1161c0, 0xc820081280, 0x0, 0x0)
	/home/philip/.gvm/gos/go1.5/src/net/http/server.go:1765 +0x85
github.com/influxdb/influxdb/services/httpd.(*Service).serve(0xc82001ad80)
	/tmp/build/src/github.com/influxdb/influxdb/services/httpd/service.go:133 +0x6e
created by github.com/influxdb/influxdb/services/httpd.(*Service).Open
	/tmp/build/src/github.com/influxdb/influxdb/services/httpd/service.go:101 +0x6ca

goroutine 32 [select]:
github.com/influxdb/influxdb/services/retention.(*Service).deleteShardGroups(0xc82000b5e0)
	/tmp/build/src/github.com/influxdb/influxdb/services/retention/service.go:69 +0x351
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
	/tmp/build/src/github.com/influxdb/influxdb/services/retention/service.go:45 +0x1b4

goroutine 33 [select]:
github.com/influxdb/influxdb/services/retention.(*Service).deleteShards(0xc82000b5e0)
	/tmp/build/src/github.com/influxdb/influxdb/services/retention/service.go:102 +0x7a0
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
	/tmp/build/src/github.com/influxdb/influxdb/services/retention/service.go:46 +0x1d9

goroutine 34 [chan receive]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).startServerReporting(0xc8200f17a0)
	/tmp/build/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:442 +0x1eb
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Open.func1
	/tmp/build/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:378 +0x1265

goroutine 35 [select]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Command).monitorServerErrors(0xc820080380)
	/tmp/build/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:138 +0x2c5
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Command).Run
	/tmp/build/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:120 +0xeeb

goroutine 36 [select, locked to thread]:
runtime.gopark(0xd19ec8, 0xc8202a4728, 0xbd5940, 0x6, 0x431018, 0x2)
	/home/philip/.gvm/gos/go1.5/src/runtime/proc.go:185 +0x163
runtime.selectgoImpl(0xc8202a4728, 0x0, 0x18)
	/home/philip/.gvm/gos/go1.5/src/runtime/select.go:392 +0xa64
runtime.selectgo(0xc8202a4728)
	/home/philip/.gvm/gos/go1.5/src/runtime/select.go:212 +0x12
runtime.ensureSigM.func1()
	/home/philip/.gvm/gos/go1.5/src/runtime/signal1_unix.go:227 +0x353
runtime.goexit()
	/home/philip/.gvm/gos/go1.5/src/runtime/asm_amd64.s:1696 +0x1

goroutine 47 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc820086a80, 0xc8202e3500)
	/tmp/build/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x567
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
	/tmp/build/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x627
````

**Comment by esecules** [comment]
is this slated for a patch to 0.9.4 or has it been bumped to 0.9.5?

**Comment by lfittl** [comment]
Running into the same issue - from reading @beckettsean's comment in #4326 this won't be fixed on the 0.9.4 branch, but rather be implemented correctly with the new storage engine in 0.9.5.

The last blog post also mentions this as "The 0.9.5 release will ship with this new storage engine along with the ability to do hot backups against it." https://influxdb.com/blog/2015/10/07/the_new_influxdb_storage_engine_a_time_structured_merge_tree.html
